compiler/src/dotty/tools/dotc/core/PatternTypeConstrainer.scala

```diff
@@ -143,10 +143,10 @@ trait PatternTypeConstrainer { self: TypeComparer =>
               ctx.gadt.bounds(scrut.symbol) match {
                 case tb: TypeBounds =>
                   val hi = tb.hi
-                  constrainPatternType(pat, hi)
-                case null => false
+                  tryConstrainSimplePatternType(pat, hi)
```

Comment:
For GADT bounds we should call `constrainPatternType` instead of `tryConstrainSimplePatternType` because the reconstructed/GADT upper bound may not be a simple type (it may be an intersection).

Reply:
:+1: I confused `tryConstrainSimplePatternType`/`constrainSimplePatternType` with `constrainPatternType`